owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.kt

```diff
@@ -0,0 +1,117 @@
+*   furnished to do so, subject to the following conditions:
+*
+*   The above copyright notice and this permission notice shall be included in
+*   all copies or substantial portions of the Software.
+*
+*   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+*   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+*   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+*   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+*   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+*   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+*   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+*   THE SOFTWARE.
+*
+*/
+package com.owncloud.android.lib.resources.users
+
+import at.bitfire.dav4android.Property
+import at.bitfire.dav4android.property.QuotaAvailableBytes
+import at.bitfire.dav4android.property.QuotaUsedBytes
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.http.HttpConstants
+import com.owncloud.android.lib.common.http.methods.webdav.DavConstants
+import com.owncloud.android.lib.common.http.methods.webdav.DavUtils
+import com.owncloud.android.lib.common.http.methods.webdav.PropfindMethod
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.operations.RemoteOperationResult.ResultCode
+import com.owncloud.android.lib.resources.users.GetRemoteUserQuotaOperation.RemoteQuota
+import timber.log.Timber
+import java.net.URL
+import kotlin.math.roundToLong
+
+/**
+ * @author marcello
+ * @author David González Verdugo
+ */
+class GetRemoteUserQuotaOperation : RemoteOperation<RemoteQuota>() {
+    override fun run(client: OwnCloudClient): RemoteOperationResult<RemoteQuota> {
+        lateinit var result: RemoteOperationResult<RemoteQuota>
+        try {
+            val propfindMethod = PropfindMethod(
+                URL(client.userFilesWebDavUri.toString()),
+                DavConstants.DEPTH_0,
+                DavUtils.getQuotaPropSet()
+            )
+            val status = client.executeHttpMethod(propfindMethod)
+            if (isSuccess(status)) {
+                val remoteQuota = readData(propfindMethod.root.properties)
+                result = RemoteOperationResult<RemoteQuota>(ResultCode.OK).apply {
+                    data = remoteQuota
+                }
+                Timber.i("Get quota completed: ${result.data} and message: ${result.logMessage}")
+
+            } else { // synchronization failed
+                result = RemoteOperationResult(propfindMethod)
+                Timber.e("Get quota without success: ${result.logMessage}")
+            }
+        } catch (e: Exception) {
+            result = RemoteOperationResult(e)
+            Timber.e(result.exception, "Get quota: ${result.logMessage}")
+        }
+        return result
+    }
```

Comment:
```suggestion
override fun run(client: OwnCloudClient): RemoteOperationResult<RemoteQuota> =
    try {
        val propfindMethod = PropfindMethod(
            URL(client.userFilesWebDavUri.toString()),
            DavConstants.DEPTH_0,
            DavUtils.getQuotaPropSet()
        )
        with(client.executeHttpMethod(propfindMethod)) {
            if (isSuccess(this)) {
                RemoteOperationResult<RemoteQuota>(ResultCode.OK).apply {
                    data = readData(propfindMethod.root.properties)
                }.also {
                   Timber.i("Get quota completed: ${it.data} and message: ${it.logMessage}")
                }
            } else { // synchronization failed
                RemoteOperationResult<RemoteQuota>(propfindMethod).also {
                    Timber.e("Get quota without success: ${it.logMessage}")
                }
            }
        }
    } catch (e: Exception) {
        RemoteOperationResult<RemoteQuota>(e).also {
            Timber.e(it.exception, "Get quota: ${it.logMessage}")
        }
    }
```

Reply:
That is a version with scoped functions and expression body, just a suggestion in case you like it... 🍺